pkg/sql/show_tenant.go

```diff
@@ -93,100 +110,145 @@ func (n *showTenantNode) getTenantName(params runParams) (roachpb.TenantName, er
 }
 
 func (n *showTenantNode) startExec(params runParams) error {
-	tenantName, err := n.getTenantName(params)
+	if n.all {
+		names, err := GetAllTenantNames(params.ctx, params.p.execCfg, params.p.Txn())
```

Comment:
I'm wondering, instead of passing this list of names around. Could we do more of this in sql?  That is, we are essentially pretty-printing the results of `SELECT id,  FROM system.tenants` and annotating it with job details. I wonder if we could either: (1) use an iterator over a query we construct to select the correct nodes or (2) implement this as a delegated statement instead.

I'm OK doing this as a follow-up, just a _little_ suspect of anywhere that we require passing around complete sets of user-generatable data.